torch/distributed/fsdp/_optim_utils.py

```diff
@@ -1269,3 +1290,36 @@ def _optim_state_dict(
         )
 
     return fsdp_osd
+
+
+def _get_fqn_to_fsdp_param_info(
+    model: nn.Module, dedup_shared_fqns: Set[str]
+) -> Dict[str, FSDPParamInfo]:
+    def module_fn(module, prefix, fqn_to_param_info):
+        # TODO: make it work with composable API.
+        if not isinstance(module, fsdp_file.FullyShardedDataParallel):
+            return
+        _lazy_init(module, module)
+        handles = _module_handles(module, module)
+        if not handles:
+            return
+        flat_param = handles[0].flat_param
+        fsdp_param_info = FSDPParamInfo(module, flat_param, {})
+        for idx, local_fqn in enumerate(flat_param._fqns):
+            fqn = clean_tensor_name(prefix + local_fqn)
+            if fqn in fqn_to_param_info:
+                assert fqn_to_param_info[fqn].flat_param == flat_param
+            if fqn in dedup_shared_fqns:
+                fqn_to_param_info[fqn] = fsdp_param_info
+                fsdp_param_info.fqn_indices[fqn] = idx
+
+    def return_fn(fqn_to_param_info):
+        return fqn_to_param_info
+
+    fqn_to_param_info: Dict[str, FSDPParamInfo] = {}
+    return _apply_to_modules(
+        model,
+        module_fn,
+        return_fn,
+        fqn_to_param_info,
+    )
```

Comment:
nit: Could you add to the PR description that the helper you added is `_get_fqn_to_fsdp_param_info` just to make that explicit?